### ISSUE TITLE
API: Fetching endpoint /api/articles/me sometimes returns CORS error.

### ISSUE THREAD

**Issue by emilpriver** [forem/forem#5877]
Currently working with the api and I see that the endpoint gives CORS error, but on second request does it work. This do i only get while using /api/articles/me request.

![4195f9c558aec8c1ba4be778f81c5be3](https://user-images.githubusercontent.com/34039674/73696730-2e7ecc80-46dd-11ea-8c3d-d675ba8fdb2a.png)

**To Reproduce**

1. use Axios GET
2. request https://dev.to/api/articles/me/ with api-key header with React router
3. if fails, reload and it should return a 200 OK.

**Expected behavior**

To render 200 OK response

**Desktop (please complete the following information):**

- OS: Windows 10
- Browser: Chrome
- Version: Chrome 79.

**Comment by citizen428** [comment]
Thanks for the issue! We'll take your request into consideration and follow up if we decide to tackle this issue.

To our amazing contributors: please wait until we add a `help-wanted` label to the issue before starting to work on it.

To claim an issue to work on, please leave a comment. If you've claimed the issue and need help, please ping @maestromac or @rhymes or @citizen428 from the DEV team and we will follow up within 3 business days.

For full info on how to contribute, please check out our [contributors guide](https://github.com/thepracticaldev/dev.to/blob/master/CONTRIBUTING.md).

**Comment by rhymes** [comment]
Hi @emilpriver, can you share which headers you sent?

**Comment by emilpriver** [comment]
This is the headers sent when the request fails:

![e864741db364c8b03ca066d98fc8a55f](https://user-images.githubusercontent.com/34039674/73770259-6557ef80-477c-11ea-814a-2419ee345a0d.png)

And this is the request which is sent when it success:
![40c47b7b0ceba3de561318f680969528](https://user-images.githubusercontent.com/34039674/73770468-c67fc300-477c-11ea-9634-897ae1a449f0.jpg)

The only different i see when the request is changed is that axios adds put,post and so on. else both headers should be the same. 

This is happening when i change route using Nextjs, mby next is the cause of this problem?

**Comment by rhymes** [comment]
As axios is just a HTTP client, if we can get it to the simplest HTTP call it does in the end we might be able to reproduce it. 

For example, I opened https://dev.to in Firefox, opened the Developer Tools's Console and wrote this:

```javascript
await fetch("https://dev.to/api/articles")
```

I then opened the network panel (which I cleared first) and then used the following menu to copy the actual HTTP call:

![Screenshot 2020-02-04 at 6 41 21 PM](https://user-images.githubusercontent.com/146201/73771215-3c385e80-477e-11ea-805c-1433cd0cf6ab.png)

If you can paste here the content of "copy as cURL" or "copy as fetch" or "copy request headers" it'd be awesome!

Don't forget to hide your API key and other private data :)

**Comment by emilpriver** [comment]
fetch("https://dev.to/api/articles/me/", {"credentials":"omit","headers":{"accept":"application/json, text/plain, */*","api-key":"HIDDEN"},"referrer":"http://localhost:3000/posts","referrerPolicy":"no-referrer-when-downgrade","body":null,"method":"GET","mode":"cors"});

Does this work?

**Comment by rhymes** [comment]
Thank you @emilpriver, I was able to reproduce it, even from the console with this:

```js
await fetch("https://dev.to/api/articles/me/", {
  "headers": {
    "api-key":"API_KEY"
  },
  "method":"GET",
  "mode":"cors"
});
```

this is the output:

```text
Cross-Origin Request Blocked: The Same Origin Policy disallows reading the remote resource at https://dev.to/api/articles/me/. (Reason: CORS preflight response did not succeed).

Cross-Origin Request Blocked: The Same Origin Policy disallows reading the remote resource at https://dev.to/api/articles/me/. (Reason: CORS request did not succeed)
```

Our API has some known CORS issues, see https://github.com/thepracticaldev/dev.to/issues/4195

**Comment by emilpriver** [comment]
@rhymes ah got ya! Checked out the issue you sendt, I will check if i amble to help in anyway :D

**Comment by medaminefh** [comment]
Is there any updates on this ?
I'm facing the same issue here

**Comment by tarunfy** [comment]
<img width="389" alt="Screenshot 2022-05-12 at 3 44 30 PM" src="https://user-images.githubusercontent.com/51208270/168048305-e1a5361f-6586-4964-bd44-a3056df388e7.png">

having CORS error while making a simple GET request at **"https://dev.to/api/articles/me/published",** can someone help me with this? 😕

Here's the code

<img width="1146" alt="Screenshot 2022-05-12 at 3 47 23 PM" src="https://user-images.githubusercontent.com/51208270/168048871-1e200b1c-7014-4a75-ba9f-12b0685dca39.png">

**Comment by tarunfy** [comment]
<img width="388" alt="Screenshot 2022-05-12 at 3 54 04 PM" src="https://user-images.githubusercontent.com/51208270/168050131-e87c5534-aa62-4436-9ac1-4f6e295ea0b9.png">

Getting CORS error in simple get request at **"https://dev.to/api/articles/me/published"**

Down below is the code I wrote:
<img width="1194" alt="Screenshot 2022-05-12 at 3 54 32 PM" src="https://user-images.githubusercontent.com/51208270/168050346-37e13c2e-56e0-4326-b61f-da5e65606ae3.png">

**Comment by mikeydorje** [comment]
**Reviewed: Closing as appears to be not in scope.**

This seems to be an intentional design decision rather than a bug. PR #6116 (Feb 2020) established that authenticated API calls with `api-key` headers appear to be intentionally not supported from browser contexts, likely to prevent users from accidentally exposing API keys in frontend code.

Related issue #16751 (Mar 2022) reported similar concerns and was converted to a discussion in Sep 2023, which suggests this may not be moving forward.

**Suggested workaround:** For authenticated API calls from a frontend, consider using a backend proxy server that handles authentication server-side.
